node/src/blockchain/blockchain_bridge.rs

```diff
@@ -653,13 +455,171 @@ mod tests {
+
+        let _ = addr.try_send(ReportAccountsPayable {
+            accounts: vec![
+                PayableAccount {
+                    wallet: make_wallet("blah"),
+                    balance: 420,
+                    last_paid_timestamp: from_time_t(150_000_000),
+                    pending_payment_transaction: None,
+                },
+                PayableAccount {
+                    wallet: make_wallet("foo"),
+                    balance: 210,
+                    last_paid_timestamp: from_time_t(120_000_000),
+                    pending_payment_transaction: None,
+                },
+            ],
+        });
+        System::current().stop();
+        system.run();
+
+        let send_transaction_params = send_transaction_params_arc.lock().unwrap();
+        assert_eq!(
+            *send_transaction_params,
+            vec![
+                (
+                    consuming_wallet.clone(),
+                    make_wallet("blah"),
+                    420,
+                    U256::from(1),
+                    expected_gas_price
+                ),
+                (
+                    consuming_wallet.clone(),
+                    make_wallet("foo"),
+                    210,
+                    U256::from(2),
+                    expected_gas_price
+                )
+            ]
+        );
+        let get_transaction_count_params = get_transaction_count_params_arc.lock().unwrap();
+        assert_eq!(
+            *get_transaction_count_params,
+            vec![consuming_wallet.clone(), consuming_wallet.clone()]
+        );
+        let accountant_received_payment = accountant_recording_arc.lock().unwrap();
+        assert_eq!(accountant_received_payment.len(), 1);
+        let sent_payment_msg = accountant_received_payment.get_record::<SentPayments>(0);
+        let mut sent_payments_cloned = sent_payment_msg.payments.clone();
+        sent_payments_cloned
+            .iter_mut()
+            .for_each(|payment_in_result| {
+                if let Ok(payment) = payment_in_result {
+                    payment.timestamp = from_time_t(0)
+                }
+            });
+        assert_eq!(
+            sent_payments_cloned,
+            vec![
+                Ok(Payment {
+                    to: make_wallet("blah"),
+                    amount: 420,
+                    timestamp: from_time_t(0), //cannot be asserted directly, this field is just a sentinel
+                    transaction: H256::from("sometransactionhash".keccak256())
+                }),
+                Ok(Payment {
+                    to: make_wallet("foo"),
+                    amount: 210,
+                    timestamp: from_time_t(0), //cannot be asserted directly, this field is just a sentinel
+                    transaction: H256::from("someothertransactionhash".keccak256())
+                })
+            ]
+        )
+    }
+
+    #[test]
+    fn retrieve_transactions_sends_transactions_to_blockchain_interface() {
```

Comment:
A better name, mentioning that we're sending a message at the end of the test (and whom), would be becoming, this one says just a half of the story.